### PR TITLE
blockdev_commit_cancel:cancel a commit job

### PR DESCRIPTION
**Pull request by aliang123**
New case:Cancel a commit job
id: 3151

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u6.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.6.0.x86_64.io-github-autotest-qemu.blockdev_commit_cancel.q35: PASS (74.15 s)

**Comment by aliang123** [comment]
@zhencliu @XueqiangWei @maxujun , please help review it, thanks.

**Comment by maxujun** [comment]
Suggest adding required_qemu required_qemu = [9.1.0, ) ,it' not suitable if qemu is below 9.1 version.

File: qemu/tests/blockdev_commit_cancel.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                self.test.fail("Commit job failed to cancel")
+
+    def check_backing_rw(self):
+        base_image = self.get_image_by_tag(self.params["device_tag"])
+        cmd = "lsof %s" % base_image.image_filename
+        output = process.run(cmd, verbose=True).stdout_text.split("\n")[1].split()
+        pid, fd = (output[1], output[3][:-1])
+        cmd = "cat /proc/%s/fdinfo/%s" % (pid, fd)
+        output = process.run(cmd, verbose=True).stdout_text.split("\n")[1].split()[1]
+        if output[-1] == "2":
```
Comment (zhencliu) [inline qemu/tests/blockdev_commit_cancel.py:46]: do we still need to check other invalid values here?

File: provider/virt_storage/storage_volume.py [hunk 2]
```diff
@@ -157,8 +157,11 @@ def refresh_format_by_params(self, params):
 
             backing = params.get("backing")
             if backing:
-                backing_node = "drive_%s" % backing
-                self.format.set_param("backing", backing_node)
+                if params.get("backing_null", "no") == "no":
```
Comment (zhencliu) [inline provider/virt_storage/storage_volume.py:160]: It's better to use another commit for the changes in provider

Comment (zhencliu) [reply provider/virt_storage/storage_volume.py:160]: But for this patch, it's up to you, I am also OK. For the patches pushed in future, please split it into two commits in one patch :-D

**Comment by aliang123** [comment]
For nbd:
INFO | Command 'lsof -i:10831' finished with 0 after 0.338442080s
DEBUG| [stdout] qemu-kvm 135158 root   19u  IPv4 563189      0t0  TCP dell-per660-xx.lab.eng.pek2.redhat.com:60478->dell-per740xd-01.lab.eng.pek2.redhat.com:10831 (ESTABLISHED)

For ceph:
INFO | Command 'lsof -i:6800' finished with 0 after 0.331290133s
DEBUG| [stdout] qemu-kvm 136048 root   28u  IPv4 561978      0t0  TCP dell-per660-xx.lab.eng.pek2.redhat.com:40466->ibm-x3650m4-06.lab.eng.pek2.redhat.com:6800 (ESTABLISHED)

File: provider/virt_storage/storage_volume.py (hunk 2 above)
Comment (aliang123) [reply provider/virt_storage/storage_volume.py:160]: Have splited the commit, thanks @zhencliu

File: qemu/tests/cfg/blockdev_commit_cancel.cfg [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        enable_iscsi_sn4 = no
+        image_raw_device_sn4 = no
+        enable_iscsi_sn11 = no
+        image_raw_device_sn11 = no
+        enable_iscsi_sn21 = no
+        image_raw_device_sn21 = no
+        enable_iscsi_sn31 = no
+        image_raw_device_sn31 = no
+        enable_iscsi_sn41 = no
+        image_raw_device_sn41 = no
```
[the remark this answers is not in the document]
Comment (aliang123) [reply qemu/tests/cfg/blockdev_commit_cancel.cfg]: Have removed the sn*, thanks @zhencliu

File: qemu/tests/blockdev_commit_cancel.py (hunk 1 above)
Comment (aliang123) [reply qemu/tests/blockdev_commit_cancel.py:50]: Yes, checking the last number is enough here for rw/ro.

**Comment by zhencliu** [comment]
Talked with @aliang123 , we will keep the hard code 6800 inline currently until we find the right way to pass the valid port

**Review by zhencliu — DISMISSED** [review]

File: qemu/tests/blockdev_commit_cancel.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockDevCommitCancel(BlockDevCommitTest):
+    def generate_tempfile(self, root_dir, filename="data", size="1500M", timeout=360):
+        backup_utils.generate_tempfile(self.main_vm, root_dir, filename, size, timeout)
+        self.files_info.append([root_dir, filename])
+
+    def commit_snapshot(self):
+        device = self.params["device_tag"].split()[0]
```
Comment (XueqiangWei) [inline qemu/tests/blockdev_commit_cancel.py:13]: Built in method get_list is functional. self.params.get_list("device_tag")[0]

File: qemu/tests/blockdev_commit_cancel.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def run(test, params, env):
+    """
+    Block commit base Test
+
+    1. boot guest with data disk
+    2. create 4 snapshots and save file in each snapshot
+    3. commit snapshot 3 to snapshot 4
+    6. verify files's md5
```
Comment (XueqiangWei) [inline qemu/tests/blockdev_commit_cancel.py:70]: here is step 4 instead of 6.

**Review by XueqiangWei — APPROVED** [review]
LGTM,
Acked-by: xuwei@redhat.com

**Pull request merged**